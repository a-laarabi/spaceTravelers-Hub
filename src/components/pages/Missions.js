import { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { LoadMissions } from '../../redux/missions/mission';
import MissionProp from './MissionProp';
import '../../styles/mission.css';

const Missions = () => {
  const missions = useSelector((state) => state.missions);
  const dispatch = useDispatch();

  useEffect(() => {
    if (missions.length === 0) {
      dispatch(LoadMissions());
    }
  }, []);

  return (
    <div className="missionTable">
      <table>
        <tbody>
          <tr>
            <th>Mission</th>
            <th>Description</th>
            <th>Status</th>
            <th> </th>
          </tr>
          {missions.map((mission) => (
            <MissionProp
              key={mission.id}
              id={mission.id}
              name={mission.name}
              description={mission.description}
              reserved={mission.reserved}
            />
          ))}
        </tbody>
      </table>
    </div>
  );
};

export default Missions;
