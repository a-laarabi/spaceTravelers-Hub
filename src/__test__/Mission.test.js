import React from "react";
import store from "../redux/configureStore";
import Missions from "../components/pages/Missions";
import { Provider } from "react-redux";
import renderer from 'react-test-renderer';

describe('Test for Missions components', () => {
  it('Check if the Missions section are available', () => {
    const mission = renderer.create(
      <Provider store={store}>
        <Missions />
      </Provider>,
    );
  expect(mission).toMatchSnapshot();
  })
})