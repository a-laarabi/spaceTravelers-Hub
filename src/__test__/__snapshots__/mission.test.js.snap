// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for Missions components Check if the Missions section are available 1`] = `
<div
  className="missionTable"
>
  <table>
    <tbody>
      <tr>
        <th>
          Mission
        </th>
        <th>
          Description
        </th>
        <th>
          Status
        </th>
        <th>
           
        </th>
      </tr>
    </tbody>
  </table>
</div>
`;
